### ISSUE TITLE
Create a Contribution guide

### ISSUE THREAD

**Issue by danicat** [sourcegraph/go-langserver#315]
I propose the creation of a CONTRIBUTING.md file or something similar to include the guidelines for contributing for this project.

It would be desirable to include a step-by-step setup guide and the contribution workflow (ie.: fork + pull request).

I can work on this if you want. :)

**Comment by slimsag** [comment]
What did you have in mind? (maybe link to an existing contributing file you like?)

In general, we don't have any special contribution rules or anything -- it's the same as almost every other OSS project:

- PRs welcome.
- If you want to avoid working on a PR that may not initially be accepted (e.g. for complex tasks), file an issue to discuss first.

I'd be hesitant to add a file that just outlines that, but am curious to hear what you think we could improve here! 😃

**Comment by danicat** [comment]
You got the idea. I'm thinking about working on some issues and it would be useful to have some guidelines on where to start (ie.: setting up the environment for development). Now that I'm thinking about it, maybe the CONTRIBUTING.md alone would not be enough. Usually this file has some overall guidelines on how to start and what are the best practicies to follow. But the environment setup maybe would fit better as an expansion of the README.md as you said.

Examples of some nice CONTRIBUTING.md files:
https://github.com/docker/cli/blob/master/CONTRIBUTING.md
https://github.com/kubeflow/kubeflow/blob/master/CONTRIBUTING.md

The best one (Which actually points out to an entire repo describing the process):
https://github.com/kubernetes/kubernetes/blob/master/CONTRIBUTING.md
(https://github.com/kubernetes/community/tree/master/contributors/guide)

**Comment by danicat** [comment]
Of course, we wouldn't need something so big as the Kubernetes Contributors Guide, but having some sections describing best practices and environment setup would be nice.

**Comment by slimsag** [comment]
Got it, that makes sense. 😃

> Of course, we wouldn't need something so big as the Kubernetes Contributors Guide, but having some sections describing best practices and environment setup would be nice.

I very much agree! I think for now we can probably just write this information into the README. If it becomes too large or needs to become more detailed, then it can later be moved to a contributing file.
